File: src/main/java/jb/files/serialization/ModelWriter.java
package jb.files.serialization;

import de.bwaldvogel.liblinear.Linear;
import de.bwaldvogel.liblinear.Model;
import jb.config.Opts;

import java.io.File;
import java.io.IOException;
import java.util.List;

import static jb.config.Constants.modelsPath;

public class ModelWriter {

    public void saveModels(List<Model> clfs, Opts opts) {

        for (int i = 0; i < clfs.size(); i++) {
            try {
                Linear.saveModel(new File(modelsPath + "/" + opts.getFilename().substring(0, 3) + "_" + i + "_on_" + clfs.size()), clfs.get(i));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

    }

}

File: src/main/kotlin/jb/files/serialization/ModelWriter.kt
package jb.files.serialization

import de.bwaldvogel.liblinear.Linear
import de.bwaldvogel.liblinear.Model
import jb.config.Constants.Companion.modelsPath
import jb.config.Opts

import java.io.File
import java.io.IOException

class ModelWriter {

    fun saveModels(clfs: List<Model>, opts: Opts) {

        for (i in clfs.indices) {
            try {
                val root = File(modelsPath)
                if (!root.exists()) root.mkdir()
                Linear.saveModel(File(modelsPath + "/" + opts.getFilename().substring(0, 3) + "_" + i + "_on_" + clfs.size), clfs[i])
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }

    }

}
